assets/js/modules/analytics/util/account.test.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * Account helper tests.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getAccountDefaults } from './account';
+
+describe( 'getAccountDefaults', () => {
+	const siteURL = 'https://example.com/';
+	const siteName = 'Example Site';
+	const timezone = 'Europe/Kiev';
+
+	describe( 'accountName', () => {
+		it( 'should be equal to siteName when siteName is not empty', () => {
+			expect( getAccountDefaults( { siteName, siteURL, timezone } ).accountName ).toBe( 'Example Site' );
+		} );
+
+		it( 'should be the domain name of the siteURL when siteName is falsy', () => {
+			expect( getAccountDefaults( { siteName: '', siteURL, timezone } ).accountName ).toBe( 'example.com' );
+		} );
+	} );
+
+	describe( 'propertyName', () => {
+		it( 'should be just domain name when the path of the siteURL is just a forward slash', () => {
+			expect( getAccountDefaults( { siteName, siteURL: 'https://example.com/', timezone } ).propertyName ).toBe( 'example.com' );
+		} );
+
+		it( 'should be domain name + path when the siteURL contains non-empty path', () => {
+			expect( getAccountDefaults( { siteName: '', siteURL: 'https://example.com/subsite-slug/', timezone } ).propertyName ).toBe( 'example.com/subsite-slug' );
+		} );
+
+		it( 'should throw an error if siteURL is invalid', () => {
+			expect( () => getAccountDefaults( { siteName, siteURL: undefined, timezone } ) ).toThrow( 'a valid siteURL is required.' );
+		} );
+	} );
+
+	describe( 'profileName', () => {
+		it( 'should be "All Web Site Data"', () => {
+			expect( getAccountDefaults( { siteName, siteURL, timezone } ).profileName ).toBe( 'All Web Site Data' );
+		} );
+	} );
+
+	describe( 'countryCode', () => {
+		it( 'should be equal to UA when timezone is Europe/Kiev', () => {
+			expect( getAccountDefaults( { siteName, siteURL, timezone } ).countryCode ).toBe( 'UA' );
+		} );
+	} );
+
+	describe( 'timezone', () => {
+		it( 'should be the same as provided', () => {
+			expect( getAccountDefaults( { siteName, siteURL, timezone } ).timezone ).toBe( 'Europe/Kiev' );
+		} );
+
+		it( 'should use a local timezone when the provided timezone does not have an entry in countryCodesByTimezone', () => {
+			expect( getAccountDefaults( { siteName, siteURL, timezone: 'UTC' } ).timeZone ).not.toBe( 'UTC' );
+		} );
```

Comment:
While this test passes, it doesn't quite work as expected. In this case `timeZone` is actually `undefined` because it doesn't exist on the returned object (it should be `timezone` instead). In this case, it would be preferable to expect a particular value but that isn't really possible using the normal browser-based default.

What we can do instead is update the utility to accept the fallback timezone as a second test-only parameter for DI. Normally I would prefer mocking in this case, but we would have to mock several nested function calls and isn't really necessary.

Everything else looks good here so I'll push these final changes and then this will be good to go. As a result of this, we no longer need to add the internationalization package or change the way the tests run so that can be rolled back as well. We can definitely revisit this in a new issue if needed, but unless we can force a particular locale programmatically then I'm not sure adding this data will be of much use as the resolved values seem to be location-sensitive.